### PR TITLE
fix: attempt to force latest build number for compilers

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge-admin rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -55,7 +55,7 @@ requirements:
   build:
      # ensure that all variants of cuda_compiler_version are tested
     - {{ compiler('cuda') }}              # [cuda_compiler_version != "None"]
-    - {{ compiler('c') }}                 # [cuda_compiler_version != "None"]
+    - {{ c_compiler }}_{{ target_platform }} {{ c_compiler_version }} *_4             # [cuda_compiler_version != "None"]
```
Comment (jakirkham) [inline recipe/meta.yaml:58]: Guessing we would want something like this (as Windows wouldn't need this pin)

```suggestion
    - {{ c_compiler }}_{{ target_platform }} {{ c_compiler_version }} *_4             # [linux and cuda_compiler_version != "None"]
    - {{ compiler('c') }}                                                                # [(not linux) and cuda_compiler_version != "None"]
```

Comment (beckermr) [reply recipe/meta.yaml:58]: Yes but this is just for testing. If this works, we'll want to use a repodata patch.

**Comment by beckermr** [comment]
ouch another rerender error. I suspect something is still broken in how the latest conda-build versions inject variables into the jinja2 templates.

**Comment by beckermr** [comment]
@conda-forge-admin rerender

File: recipe/meta.yaml [hunk 2]
```diff
@@ -21,9 +21,9 @@ build:
   skip: True
   {% endif %}
   script:
-    - cp {{ os.environ["FEEDSTOCK_ROOT"] }}/LICENSE.txt ${RECIPE_DIR}/LICENSE.txt            # [linux]
-    - cp ${RECIPE_DIR}/../LICENSE.txt ${RECIPE_DIR}/LICENSE.txt                              # [osx]
-    - COPY "%RECIPE_DIR%\\..\\LICENSE.txt" "%RECIPE_DIR%\\LICENSE.txt"                       # [win]
+    - cp {{ os.environ.get("FEEDSTOCK_ROOT", "failme") }}/LICENSE.txt ${RECIPE_DIR}/LICENSE.txt  # [linux]
```
Comment (jakirkham) [inline recipe/meta.yaml:25]: We can just guard the usage if needed

```suggestion
    {% if "FEEDSTOCK_ROOT" in os.environ %}
    - cp {{ os.environ["FEEDSTOCK_ROOT"] }}/LICENSE.txt ${RECIPE_DIR}/LICENSE.txt  # [linux]
    {% endif %}
```

Comment (beckermr) [reply recipe/meta.yaml:25]: Let's move this comment to PR #350

**Comment by beckermr** [comment]
@conda-forge-admin rerender

File: recipe/meta.yaml (hunk 2 above)
Comment (jakirkham) [reply recipe/meta.yaml:25]: Only responded here as we were having re-rendering issues: https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/351#issuecomment-2375527805

Comment (beckermr) [reply recipe/meta.yaml:25]: yeah that is a new one that likely has the same underlying cause but is for a different variable...

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by jakirkham** [comment]
Looks promising 🙂

**Comment by jakirkham** [comment]
Continuing discussion in issue: https://github.com/conda-forge/binutils-feedstock/issues/78

**Comment by beckermr** [comment]
Oh fascinating. These builds with the c compiler included appear to pass. OTOH these builds failed: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1037636&view=logs&j=09c6ee00-92ab-5751-f80f-c049ce8ce0ed&t=7a4216ee-c61d-5132-71cd-0d47b3b4b519

**Comment by jakirkham** [comment]
Repeating this comment ( https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/349#issuecomment-2375439341 ) to keep everything in this thread

We noticed the `conda-forge-ci-setup` cross-compiled architectures are failing (for [example]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1037636&view=logs&j=09c6ee00-92ab-5751-f80f-c049ce8ce0ed&t=7a4216ee-c61d-5132-71cd-0d47b3b4b519&l=866 ))

```python
__main__.py: error: argument --sysroot: expected one argument
```

This led us to the `cross-python` package, which has this logic in [its activation script]( https://github.com/conda-forge/cross-python-feedstock/blob/d94169b29f29de1b24dc39576ac675dac8a6d048/recipe/activate-cross-python.sh#L32-L38 ):

```bash
    $BUILD_PREFIX/bin/python -m crossenv $PREFIX/bin/python \
        --sysroot $CONDA_BUILD_SYSROOT \
        --without-pip $BUILD_PREFIX/venv \
        --sysconfigdata-file "$sysconfigdata_fn" \
	--machine ${machine} \
        --cc ${CC:-@CC@} \
        --cxx ${CXX:-@CXX@}
```

So for some reason `$CONDA_BUILD_SYSROOT` is now undefined

Certainly if we add the compilers (as done here), [`CONDA_BUILD_SYSROOT` will be set by the compilers]( https://github.com/conda-forge/ctng-compiler-activation-feedstock/blob/d415477224e3e7bb411766c0e488964e481ab7c8/recipe/activate-gcc.sh#L187 )

```bash
  "CONDA_BUILD_SYSROOT,${CONDA_PREFIX}@LIBRARY_PREFIX@/@CHOST@/sysroot" \
```

However that means all users of `cross-python` now need to add the compilers as a build dependency (if they don't already). Is this an expected/intended change?

**Pull request closed without merging**